### PR TITLE
IGNITE-15102 Add handling events.

### PR DESCRIPTION
**Pull request by ivandasch**

### CODE REVIEW

File: pyignite/monitoring.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+    def on_query_fail(self, event: QueryFailEvent):
+        """
+        Handle query fail event.
+
+        :param event: Instance of :class:`QueryFailEvent`.
+        """
+        pass
+
+
+class _EventListeners:
```
Comment (isapego) [inline pyignite/monitoring.py:390]: Is it common practice in Python that there is a separate method for every event? Why won't we just add a `event_type` field to an event and `on_event` message to listener? With current approach we need to create new methods for every new event.

**Review by isapego — COMMENTED** [review]
Looks good overall, but there are few questions inline.

File: pyignite/stream/aio_cluster.py [hunk 2]
```diff
@@ -1,53 +0,0 @@
...
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-"""
-This module contains `AioCluster` that lets you get info and change state of the
-whole cluster.
-"""
-from pyignite import AioClient
-from pyignite.api.cluster import cluster_get_state_async, cluster_set_state_async
-
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply pyignite/stream/aio_cluster.py:22]: I suppose you have added this by mistake and I've missed this on review. I realised that this is file is not used at all while ran test with coverage.

File: pyignite/monitoring.py (hunk 1 above)
Comment (ivandasch) [reply pyignite/monitoring.py:390]: >> With current approach we need to create new methods for every new event.
Yep, but it is much more convenient to user. Firstly, user should not create these methods by hand. Secondly,
it should not write a bunch if..elif..else code. Plus, python is a dynamic-typed language and there is not much help from IDE when you code such a thing. In current apporach these problems are eliminated.

Comment (ivandasch) [reply pyignite/monitoring.py:390]: Here are the examples of PyCharm IDE help in this case:
1. Add new event handler
Peek 2021-07-19 18-18
2. Autocomplete fields of event
Peek 2021-07-19 18-24

**Review by isapego — APPROVED** [review]
Good, ship it

**Pull request closed without merging**